tests/PathMappingRepositoryTest.php

```diff
@@ -131,6 +131,103 @@ public function testResolveFilesystemResource()
         $this->assertEquals(Path::normalize(__DIR__.'/Fixtures/dir5/sub'), $resource->getFilesystemPath());
     }
 
+    public function testMergeDirectParent()
```

Comment:
I think we should isolate fetching from adding shouldn't we? If the `add()` solve a problem by advance, we could miss it in the fetching.